data/libs/functions/lever.lua

```diff
@@ -181,4 +180,118 @@ function Lever.setStorageAllPlayers(self, key, value) -- Will set storage on all
+		{ pos = Position(33639, 32562, 13), teleport = Position(33617, 32567, 13), effect = CONST_ME_TELEPORT },
+		{ pos = Position(33640, 32562, 13), teleport = Position(33617, 32567, 13), effect = CONST_ME_TELEPORT },
+		{ pos = Position(33641, 32562, 13), teleport = Position(33617, 32567, 13), effect = CONST_ME_TELEPORT },
+		{ pos = Position(33642, 32562, 13), teleport = Position(33617, 32567, 13), effect = CONST_ME_TELEPORT }
+	},
+	specPos = {
+		from = Position(33607, 32553, 13),
+		to = Position(33627, 32570, 13)
+	},
+	condition = function(creature)
+		return true
+	end, -- If it is nil the condition will be default
+	exit = Position(33618, 32523, 15),
+	storage = Storage.Quest.U12_00.TheDreamCourts.FacelessBaneTime
+}
+]]
+
+---@generic Player
+---@param player Player
+---@param config table
+---@return boolean
+function CreateDefaultLeverBoss(player, config) -- This function is to suppress all default lever systems for boss
+	if config.playerPositions[1].pos ~= player:getPosition() then
+		return false
+	end
+
+	local spec = Spectators()
+	spec:setOnlyPlayer(false)
+	spec:setRemoveDestination(config.exit)
+	spec:setCheckPosition(config.specPos)
+	spec:check()
+
+	if spec:getPlayers() > 0 then
+		player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "There's someone fighting with " .. config.boss.name .. ".")
+		return true
+	end
+
+	local lever = Lever()
+	lever:setPositions(config.playerPositions)
+	if type(config.condition) == "function" then
+		lever:setCondition(config.condition())
+	else
+		lever:setCondition(function(creature)
+			if not creature or not creature:isPlayer() then
+				return true
+			end
+			if config.requiredLevel and creature:getLevel() < config.requiredLevel then
+				creature:sendTextMessage(MESSAGE_EVENT_ADVANCE, "All the players need to be level " .. config.requiredLevel .. " or higher.")
+				return false
+			end
+			if config.storage and creature:getStorageValue(config.storage) > os.time() then
+				local info = lever:getInfoPositions()
+				for _, v in pairs(info) do
+					local newPlayer = v.creature
+					if newPlayer then
+						newPlayer:sendTextMessage(MESSAGE_EVENT_ADVANCE, "You or a member in your team have to wait " .. string.diff(config.timeToFightAgain) .. " to face ".. config.boss.name .." again!")
+						if newPlayer:getStorageValue(config.storage) > os.time() then
+							newPlayer:getPosition():sendMagicEffect(CONST_ME_POFF)
+						end
+					end
+				end
+				return false
+			end
+			return true
+		end)
+	end
+	lever:checkPositions()
+	if lever:checkConditions() then
+		spec:removeMonsters()
+		local monster = Game.createMonster(config.boss.name, config.boss.position, true, true)
+		if not monster then
+			return true
+		end
+		lever:teleportPlayers()
+		if config.storage then
+			lever:setStorageAllPlayers(config.storage, os.time() + config.timeToFightAgain)
+		elseif config.timeToFightAgain then
+			error("Not found config.storage")
+		end
+		addEvent(function()
```

Comment:
I think that you need to check if there is another event running before calling another one. This can cause a bug when a second "team players" go to the lever and is kicked from the room by another event.

Reply:
The system checks if the players that are inside the room are the same ones that pulled the lever, so if another team enters later, it will not be taken out of the room by another event. I did this check within the addEvent itself on line 276 and 286.
Just in case I tested it and it's working as expected.